allauth/account/adapter.py

```diff
@@ -565,9 +565,26 @@ def is_safe_url(self, url):
 
         return url_has_allowed_host_and_scheme(url, allowed_hosts=allowed_hosts)
 
+    def request_password_reset(self, form, request: HttpRequest, **kwargs) -> str:
+        """
+        Method intended to be overridden in case you need to customize the logic
+        used to determine whether a user is permitted to request a password reset.
+        For example, if you are enforcing something like "social only" authentication
+        in your app, you may want to intervene here.
+
+        """
+        from allauth.account.internal import flows
+
+        email = form.cleaned_data["email"]
+        if not form.users:
+            flows.signup.send_unknown_account_mail(request, email)
+        else:
+            form._send_password_reset_mail(request, email, form.users, **kwargs)
```

Comment:
The adapter is now calling a private method of the form. The form itself is not using that method, so there is no reason to have it on the form.